src/archive.rs

```diff
@@ -97,12 +99,12 @@ fn tar(
     tar_builder.follow_symlinks(!skip_symlinks);
     // Recursively adds the content of src_dir into the archive stream
     tar_builder
-        .append_dir_all(inner_folder, &src_dir)
+        .append_dir_all(inner_folder, src_dir.as_ref())
         .map_err(|e| {
             ContextualError::IOError(
                 format!(
                     "Failed to append the content of {} to the TAR archive",
-                    &src_dir
+                    src_dir.as_ref().to_str().unwrap()
```

Comment:
I would prefer not to have this `unwrap()`. Can you choose a sensible default? I know it's very unlikely to fail, but still.

Reply:
I have no idea except using `unwrap_or("file")` If you think good, I'll change it